net_sink.go

```diff
@@ -118,8 +118,8 @@ func NewNetSink(opts ...SinkOption) FlushableSink {
 		bufSize = defaultBufferSizeTCP
 	}
 
-	s.outc = make(chan *bytes.Buffer, approxMaxMemBytes/bufSize)
-	s.retryc = make(chan *bytes.Buffer, 1) // It should be okay to limit this given we preferentially process from this over outc.
+	s.outc = make(chan *bytes.Buffer, approxMaxMemBytes/bufSize) // todo: need to understand why/how this number was chosen and probably elevate it
```

Comment:
To make threading possible forking https://github.com/lyft/gostats/pull/169#pullrequestreview-2521121502 to here . @sokada1221. 

So this doesn't restrict memory allocated, but the amount of slots per metric/string. 
If it exceeds it'll block more stats for being written until read, it would act as a batching mechanism 🤔. buffered channeled as a bit strange i think in actuality this buffer will always be full, if we would change it to normal channel with no buffer (always block), i don't think we would see an impact.